### PR TITLE
fix: admin export and import applications and applicationsets from other namespaces

### LINKED ISSUE #14596: "argocd admin export" only exports ArgoCD apps from the same namespace as where ArgoCD is deployed
Hello all,

I tried to do a backup of ArgoCD using the "argocd admin export" command as described in the document: https://argo-cd.readthedocs.io/en/stable/operator-manual/disaster_recovery/

I was able to run the backup successfully but I noticed that the backup file only contained ArgoCD apps that were deployed in the same namespace as where ArgoCD is deployed.

Does the command "argocd admin export" also support exporting ArgoCD apps from other/all namespaces https://argo-cd.readthedocs.io/en/stable/operator-manual/app-any-namespace/?

Thank you in advance!

### PR DESCRIPTION
**Pull request by ilia-medvedev**
Closes #14596

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Comment by pasha-codefresh** [comment]
@CodiumAI-Agent /review

File: cmd/argocd/commands/admin/admin.go [hunk 1]
```diff
@@ -218,3 +222,83 @@ func specsEqual(left, right unstructured.Unstructured) bool {
 	}
 	return false
 }
+
+func iterateStringFields(obj interface{}, callback func(name string, val string) string) {
```
Comment (pasha-codefresh) [inline cmd/argocd/commands/admin/admin.go]: Please remove this change

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

File: cmd/argocd/commands/admin/admin.go [hunk 2]
```diff
@@ -218,3 +222,51 @@ func specsEqual(left, right unstructured.Unstructured) bool {
...
+}
+
+const (
+	applicationsetNamespacesCmdParamsKey = "applicationsetcontroller.namespaces"
+	applicationNamespacesCmdParamsKey    = "application.namespaces"
+)
+
+// Get additional namespaces from argocd-cmd-params
+func getAdditionalNamespaces(ctx context.Context, argocdClientsets *argoCDClientsets) *argocdAdditonalNamespaces {
+	applicationNamespaces := []string{}
```
Comment (pasha-codefresh) [inline cmd/argocd/commands/admin/admin.go:238]: please change to `var applicationNamespaces []string`

File: cmd/argocd/commands/admin/admin.go [hunk 3]
```diff
@@ -218,3 +222,51 @@ func specsEqual(left, right unstructured.Unstructured) bool {
...
+	applicationsetNamespaces := []string{}
+	un, err := argocdClientsets.configMaps.Get(ctx, common.ArgoCDCmdParamsConfigMapName, v1.GetOptions{})
+	errors.CheckError(err)
+	var cm apiv1.ConfigMap
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(un.Object, &cm)
+	errors.CheckError(err)
+
+	namespacesListFromString := func(namespacelist string) []string {
```
Comment (pasha-codefresh) [inline cmd/argocd/commands/admin/admin.go:246]: lets rename namespacelist to namespaces

File: cmd/argocd/commands/admin/admin.go [hunk 4]
```diff
@@ -218,3 +222,51 @@ func specsEqual(left, right unstructured.Unstructured) bool {
...
+		listOfNamespaces := []string{}
+
+		ss := strings.Split(namespacelist, ",")
+
+		for _, s := range ss {
```
Comment (pasha-codefresh) [inline cmd/argocd/commands/admin/admin.go:251]: instead s, let put namespace, also i suggest rename ss to something

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

**Comment by pasha-codefresh** [comment]
@ilia-medvedev-codefresh please run 

`make lint-local`
`make codegen-local`

Let me know if help needed

**Comment by reggie-k** [comment]
LGTM.
As per discussion with Pasha, it would be beneficial for the "import --prune" option to have an "Are you sure" prompt, listing the number of the resources that are expected to be deleted.
Opening a separate issue for that (since adding a prompt is a breaking change for CI/other automations).

File: docs/user-guide/commands/argocd_admin_import.md [hunk 5]
```diff
@@ -11,31 +11,35 @@ argocd admin import SOURCE [flags]
...
-      --server string                  The address and port of the Kubernetes API server
-      --stop-operation                 Stop any existing operations
-      --tls-server-name string         If provided, this name will be used to validate server certificate. If this is not provided, hostname used to contact the server is used.
-      --token string                   Bearer token for authentication to the API server
-      --user string                    The name of the kubeconfig user to use
-      --username string                Username for basic authentication to the API server
-      --verbose                        Verbose output (versus only changed output)
+      --application-namespaces strings      Comma separated list of namespace globs to which import of applications is allowed.
+                                            If not provided value from 'application.namespaces' in argocd-cmd-params-cm will be used,if it's not defined only applications without an explicit namespace will be imported to the Argo CD namespace
```
Comment (pasha-codefresh) [inline docs/user-guide/commands/argocd_admin_import.md:15]: Lets remove newline

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

**Review by pasha-codefresh — APPROVED** [review]

**Pull request merged**